c2rust-analyze/src/borrowck/type_check.rs

```diff
@@ -37,18 +37,20 @@ impl<'tcx> TypeChecker<'tcx, '_> {
         )
     }
 
-    pub fn visit_place(&mut self, pl: Place<'tcx>) -> LTy<'tcx> {
-        let mut lty: LTy = self.local_ltys[pl.local.index()];
-
-        let mut adt_func = |base_lty: LTy<'tcx>, base_adt_def: AdtDef, field: Field| {
-            let base_origin_param_map: IndexMap<OriginParam, Origin> =
-                IndexMap::from_iter(base_lty.label.origin_params.to_vec());
-            let field_def: &FieldDef = &base_adt_def.non_enum_variant().fields[field.index()];
-            let perm = self.field_permissions[&field_def.did];
-            let base_metadata = &self.adt_metadata.table[&base_adt_def.did()];
-            let field_metadata = &base_metadata.field_info[&field_def.did];
-
-            self.ltcx.relabel(
+    pub fn projection_lty(
```

Comment:
Should probably be named `field_lty` or something like that, since it only works for fields, not projections in general.

Reply:
renamed in 7de4be76